### PR TITLE
Refactor: move displayname logic from backend to frontend

### PR DESCRIPTION
**Pull request by eleijonmarck**
**What is this feature?**
This removes fallbackDisplayName from the backend. This was causing inconsistent states between grafana and terraform.
Since this is to make it human readable we move the logic to the frontend.

- adds a util for adding displayname to the role
- removes backend fallbackdisplayname

**Which issue(s) does this PR fix?**:
https://github.com/grafana/grafana/issues/62680

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/core/components/RolePicker/RolePickerInput.tsx [hunk 1]
```diff
@@ -60,7 +60,7 @@ export const RolePickerInput = ({
     <div className={styles.wrapper}>
       {showBasicRole && <ValueContainer>{basicRole}</ValueContainer>}
       {appliedRoles.map((role) => (
-        <ValueContainer key={role.uid}>{role.displayName}</ValueContainer>
+        <ValueContainer key={role.uid}>{role.displayName || role.name}</ValueContainer>
```
Comment (eleijonmarck) [inline public/app/core/components/RolePicker/RolePickerInput.tsx:63]: this is not part of the move but in cases we do not have a displayname for the role this field would be empty

File: public/app/core/components/RolePicker/api.ts [hunk 2]
```diff
@@ -19,10 +22,11 @@ export const fetchUserRoles = async (userId: number, orgId?: number): Promise<Ro
     userRolesUrl += `?targetOrgId=${orgId}`;
   }
   try {
-    const roles = await getBackendSrv().get(userRolesUrl);
+    let roles = await getBackendSrv().get(userRolesUrl);
     if (!roles || !roles.length) {
       return [];
     }
+    roles = roles.map(addDisplayNameForFixedRole);
```
Comment (alexanderzobnin) [inline public/app/core/components/RolePicker/api.ts:29]: We also need to update `fetchTeamRoles()`

File: public/app/core/components/RolePicker/api.ts [hunk 3]
```diff
@@ -1,15 +1,18 @@
...
   let rolesUrl = '/api/access-control/roles?delegatable=true';
   if (orgId) {
     rolesUrl += `&targetOrgId=${orgId}`;
   }
-  const roles = await getBackendSrv().get(rolesUrl);
+  let roles = await getBackendSrv().get(rolesUrl);
   if (!roles || !roles.length) {
     return [];
   }
+  roles = roles.map(addDisplayNameForFixedRole);
```
Comment (alexanderzobnin) [inline public/app/core/components/RolePicker/api.ts:15]:
```suggestion
  return roles.map(addDisplayNameForFixedRole);
```
And then you can change `let` to `const` in declaration

File: public/app/core/components/RolePicker/utils.ts [hunk 4]
```diff
@@ -3,3 +3,19 @@ import { Role } from 'app/types';
...
 };
+
+// addDisplayNameForFixedRole provides a fallback name for fixed roles
+// this is "incase" a fixed role is introduced but without a displayname set
+// example: currently this would give:
+// fixed:datasources:name -> datasources name
+// fixed:datasources:admin      -> datasources admin
+export const addDisplayNameForFixedRole = (role: Role) => {
+  const fixedRolePrefix = 'fixed:';
+  if (role.name.startsWith(fixedRolePrefix) && !role.displayName) {
```
Comment (alexanderzobnin) [inline public/app/core/components/RolePicker/utils.ts:14]:
```suggestion
  if (!role.displayName && role.name.startsWith(fixedRolePrefix)) {
```
More clear and probably a bit cheaper (`!role.displayName` is cheaper for evaluation and if it's `false`, then the second one won't be evaluated at all).

**Review by alexanderzobnin — APPROVED** [review]
LGTM, just a few minor tweaks.

File: public/app/core/components/RolePicker/utils.ts (hunk 4 above)
Comment (eleijonmarck) [reply public/app/core/components/RolePicker/utils.ts:14]: ah, nice improvement

File: public/app/core/components/RolePicker/api.ts (hunk 2 above)
Comment (eleijonmarck) [reply public/app/core/components/RolePicker/api.ts]: [3eacfae](https://github.com/grafana/grafana/pull/62845/commits/3eacfae4566a50e4bc1d9939093976868018cba2)

File: public/app/core/components/RolePicker/api.ts (hunk 3 above)
Comment (eleijonmarck) [reply public/app/core/components/RolePicker/api.ts]: [48907cd](https://github.com/grafana/grafana/pull/62845/commits/48907cd9bab6eca9768ecc61f008330184d18d96)

**Pull request merged**